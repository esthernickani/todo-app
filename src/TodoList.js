import React, { useState } from "react";
import Todo from "./Todo"
import { v4 as uuid } from 'uuid';
import NewTodoForm from "./NewTodoForm"

const TodoList = () => {
    const [todos, setTodos] = useState([])

    const addTodo = (newtodo) => {
        setTodos(todos => [...todos, {id:uuid(), ...newtodo}])
    }

    const removeTodo = (id) => {
        const updatedTodos = todos.filter(todo => todo.id !== id)
        setTodos(updatedTodos)
    }

    const todoComponent = todos.map(({ todo, id }) => <Todo todo={todo} id={id} key={id} removeTodo={removeTodo}/>)

    return (
        <div>
            <h1>To-dos</h1>
            <ul>{todoComponent}</ul>
            <NewTodoForm addTodo={addTodo} />
        </div>
    )
}

export default TodoList;