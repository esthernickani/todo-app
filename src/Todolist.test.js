import React from "react";
import TodoList from "./TodoList";
import { fireEvent, render } from "@testing-library/react";

const addToDo = (todoList) => {
    const input = todoList.getByLabelText('Todo')
    const submitBtn = todoList.getByText('Submit')

    fireEvent.change(input, { target: {value:'This is a test todo'}})
    fireEvent.click(submitBtn)
}

it("renders without crashing", function() {
    render(<TodoList />)
})

it("matches snapshot", function() {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot()
})

it("adds todo", function() {
    const todoList = render(<TodoList />);

    //get remove button and expect it not to be in document
    const removeBtn = todoList.queryByText('X')
    expect(removeBtn).not.toBeInTheDocument();

    //add a todo
    addToDo(todoList)

    const addedTodo = todoList.queryByText('This is a test todo')
    const remove = todoList.getByText('X')

    expect(remove).toBeInTheDocument();
    expect(addedTodo).toBeInTheDocument();
})

it("deletes todo", function() {
    const todoList = render(<TodoList />);

    //add a todo
    addToDo(todoList)

    const addedTodo = todoList.queryByText('This is a test todo')
    const remove = todoList.getByText('X')

    expect(remove).toBeInTheDocument();
    expect(addedTodo).toBeInTheDocument();

    //Delete todo
    fireEvent.click(remove);
    const deletedTodo = todoList.queryByText('This is a test todo')
    expect(deletedTodo).not.toBeInTheDocument()
})