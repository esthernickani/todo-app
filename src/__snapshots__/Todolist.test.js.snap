// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      To-dos
    </h1>
    <ul />
    <form>
      <label
        for="todo"
      >
        Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Todo"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
